lib/zip/file.rb

```diff
@@ -415,15 +414,11 @@ def on_success_replace
     def create_tmpname
       dirname, basename = ::File.split(name)
       ::Dir::Tmpname.create(basename, dirname) do |tmpname|
-        opts = {perm: 0600, mode: ::File::CREAT | ::File::WRONLY | ::File::EXCL}
+        opts = {mode: ::File::CREAT | ::File::WRONLY | ::File::EXCL}
```

Comment:
Yes, my original plan to fix this was to make sure the file is created in the same place as its eventual destination - then I noticed that is exactly what is already being done!